pkg/controllers/provisioningv2/cluster/controller.go

```diff
@@ -37,7 +37,7 @@ const (
 )
 
 var (
-	mgmtNameRegexp = regexp.MustCompile("c-[a-z0-9]{5}|local")
+	mgmtNameRegexp = regexp.MustCompile("^c-[a-z0-9]{5}|local$")
```

Comment:
I guess this matches `^c-[a-z0-9]{5}` and `local$`. We want `^c-[a-z0-9]{5}$` and `^local$`, right?

```suggestion
	mgmtNameRegexp = regexp.MustCompile("^(c-[a-z0-9]{5}|local)$")
```

Reply:
good catch, thank!